torch/quantization/fx/_equalize.py

```diff
@@ -460,6 +520,21 @@ def convert_eq_obs(
             assert(isinstance(weight_eq_obs, _WeightEqualizationObserver))
             assert(isinstance(node.target, str))
 
+            if (modules[node.target] == weight_eq_obs):
+                # If the WeightEqualizationObserver is already in the node
+                # itself, this implies we are using functional layers(?) so we
+                # need to scale the weights differently
+                equalization_scale = weight_eq_obs.equalization_scale
+                scale_weight_functional(node, model, equalization_scale)
+
+                # Erase the weight equalization observer node
```

Comment:
Where do we erase the weight equalization observer for the nn.Linear case?

Reply:
For the nn.Linear case, the weight equalization observer doesn't get inserted into the graph. We create the observer in `update_obs_for_equalization` and store it in the `weight_eq_obs_dict` that gets passed into `convert_eq_obs` and use it to update the weight value, but never insert it into the graph, so we do not need to erase the weight equalization observer.